base/special/exp.jl

```diff
@@ -425,10 +431,25 @@ end
                       0.008332997481506921, 0.0013966479175977883, 0.0002004037059220124))
     small_part = r*hi
     twopk = reinterpret(Float64, (N+1023) << 52)
-    x > MAX_EXP(Float32) && return Inf32
     return Float32(muladd(twopk, small_part, twopk-1.0))
 end
 
+@inline function expm1(x::Float16)
+    x > MAX_EXP(Float16) && return Inf16
+    x < MIN_EXP(Float16) && return -one(Float16)
```

Comment:
```suggestion
    x < MIN_EXP(Float16) && return Float16(-1.0)
```
Easier to read and matches the Float32 method above?